### PR TITLE
Implement benchmarks for Oak Functions

### PR DESCRIPTION
**Pull request by tiziano88**
Remove the checked-in key value lookup Wasm module, in favour of actually building it from sources.

Remove lazy_static block in favour of referring to the necessary built crates directly.

Ref #3757
Ref #3746

### CODE REVIEW

File: oak_functions_launcher/tests/integration_test.rs [hunk 1]
```diff
@@ -56,27 +46,33 @@ async fn test_launcher_virtual() {
     ))
     .await;
 
+    let wasm_path = oak_functions_test_utils::build_rust_crate_wasm("key_value_lookup")
```
Comment (rbehjati) [inline oak_functions_launcher/tests/integration_test.rs:49]: Perhaps "key_value_lookup" and "oak_functions_linux_fd_bin" could be consts in this file?

File: oak_functions_launcher/benches/integration_benches.rs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    runtime
+        .block_on(launched_instance.kill())
+        .expect("Failed to stop launcher");
+}
+
+#[bench]
+fn bench_key_value_lookup(b: &mut Bencher) {
+    let wasm_path = oak_functions_test_utils::build_rust_crate_wasm("key_value_lookup")
+        .expect("Failed to build Wasm module");
+    run_bench(
```
Comment (rbehjati) [inline oak_functions_launcher/benches/integration_benches.rs:99]: What is being measured in this benchmark test?

**Review by rbehjati — APPROVED** [review]

**Review by andrisaar — APPROVED** [review]

File: oak_functions_launcher/benches/integration_benches.rs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+/// Similar to the integration test, but wrapped in a non-async function, and invoking the Wasm
+/// module in the benchmark loop.
+fn run_bench(b: &mut Bencher, config: &OakFunctionsTestConfig) {
+    let runtime = tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .build()
+        .unwrap();
+
+    let oak_functions_linux_fd_bin_path =
+        oak_functions_test_utils::build_rust_crate_linux("oak_functions_linux_fd_bin")
```
[the remark this answers is not in the document]
Comment (tiziano88) [reply oak_functions_launcher/benches/integration_benches.rs:50]: Yeah I totally agree. In fact I think at that point we can probably remove the linux binaries entirely, WDYT?

File: oak_functions_launcher/tests/integration_test.rs (hunk 1 above)
Comment (tiziano88) [reply oak_functions_launcher/tests/integration_test.rs:49]: I'll make some more refactorings shortly which include this, stay tuned.

File: oak_functions_launcher/benches/integration_benches.rs (hunk 2 above)
Comment (tiziano88) [reply oak_functions_launcher/benches/integration_benches.rs:99]: It is described in the rustdoc above: `[...] measuring the latency of the invocation`

**Pull request merged**